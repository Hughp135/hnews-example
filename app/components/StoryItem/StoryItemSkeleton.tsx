import { ListItem } from "../ListItem/ListItem";
import styles from "./storyitem.module.scss";

export const StoryItemSkeletonList = ({ count }: { count: number }) => {
  return (
    <>
      {Array.from({ length: count }).map((_, idx) => (
        <StoryItemSkeleton key={idx} />
      ))}
    </>
  );
};

export const StoryItemSkeleton = () => {
  return (
    <ListItem>
      <>
        <div className={styles["skeleton-text"]} />
        <div
          className={`${styles["skeleton-text"]} ${styles["skeleton-subtitle"]}`}
        />
      </>
    </ListItem>
  );
};
