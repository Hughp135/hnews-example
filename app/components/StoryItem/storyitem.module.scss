@use "../../variables" as variables;

.container {
  padding: #{variables.$spacing}px 0;

  h3 {
    margin: 0;
  }

  p {
    margin: 0;
    color: darken(variables.$lightText, 50%);
  }
}
