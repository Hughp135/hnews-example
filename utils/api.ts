export const getTopStoryIDs = async (): Promise<number[]> => {
  const res = await fetch(
    "https://hacker-news.firebaseio.com/v0/topstories.json"
  );

  if (!res.ok) {
    throw new Error("Failed to fetch top stories");
  }

  return res.json();
};

export interface Story {
  by: string;
  score: number;
  time: number;
  title: string;
  type: string;
  url: string;
}

export const getStory = async (id: number): Promise<Story> => {
  const res = await fetch(
    `https://hacker-news.firebaseio.com/v0/item/${id}.json`
  );

  if (!res.ok) {
    throw new Error("Failed to fetch story data");
  }

  return res.json();
};
