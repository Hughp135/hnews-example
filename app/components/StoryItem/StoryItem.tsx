"use client";
import { Story } from "@/utils/api";
import styles from "./storyitem.module.scss";

export const StoryItem = ({ story }: { story: Story }) => {
  return (
    <div className={styles.container}>
      <a href={story.url} target="_blank" rel="noreferrer">
        <h3>{story.title}</h3>
      </a>
      <p>
        {story.score} votes | {story.by}
      </p>
    </div>
  );
};
