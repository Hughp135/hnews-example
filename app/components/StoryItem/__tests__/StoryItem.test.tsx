import { Story } from "@/utils/api";
import { render } from "@testing-library/react";
import { StoryItem } from "../StoryItem";

describe("StoryItem", () => {
  const exampleStory: Story = {
    by: "test",
    score: 1,
    time: 201312,
    title: "A story is here",
    type: "story",
    url: "http://test.url",
  };
  it("should render the story", () => {
    const { asFragment } = render(<StoryItem story={exampleStory} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
