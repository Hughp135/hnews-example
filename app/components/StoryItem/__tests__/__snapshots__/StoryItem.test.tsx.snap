// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryItem should render the story 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <a
      href="http://test.url"
      rel="noreferrer"
      target="_blank"
    >
      <h3>
        A story is here
      </h3>
    </a>
    <p>
      1 votes | test
    </p>
  </div>
</DocumentFragment>
`;
